manager/controlapi/node.go

```diff
@@ -220,14 +240,19 @@ func (s *Server) UpdateNode(ctx context.Context, request *api.UpdateNodeRequest)
 	}
 
 	if demote && s.raft != nil {
-		memberlist := s.raft.GetMemberlist()
-		for raftID, member := range memberlist {
-			if member.NodeID == request.NodeID {
-				if err := s.raft.RemoveMember(ctx, raftID); err != nil {
-					return nil, err
+		if err := s.raft.RemoveMember(ctx, member.RaftID); err != nil {
+			// Rollback to the initial Spec if we can't finalize the role
+			// change by removing the member from raft.
+			s.store.Update(func(tx store.Tx) error {
+				node = store.GetNode(tx, request.NodeID)
+				if node == nil {
+					return nil
 				}
-				break
-			}
+
+				node.Spec = initSpec
```

Comment:
If this node is killed in the middle of this operation, won't this `initSpec` get lost, since the change is propagated by the above transaction?